staging/src/k8s.io/apiserver/pkg/storage/names/generate.go

```diff
@@ -42,13 +42,23 @@ var SimpleNameGenerator NameGenerator = simpleNameGenerator{}
 const (
 	// TODO: make this flexible for non-core resources with alternate naming rules.
 	maxNameLength          = 63
-	randomLength           = 5
-	MaxGeneratedNameLength = maxNameLength - randomLength
+	randomLengthMin        = 5
+	randomLengthMax        = 10
+	MaxGeneratedNameLength = maxNameLength - randomLengthMin // retained for compatibility
 )
 
 func (simpleNameGenerator) GenerateName(base string) string {
```

Comment:
A way to do this that might not break things is to make it variable and use the size of the collection to inform how long to make the name. We have a metric for that so it is pretty cheap to check (not sure about easy).

If there's < 5000 objects of a type, 5 characters is fine.

Reply:
I'm not convinced we need to be so clever.  This is intended not to break (at least not to fail validation) anything that works today - at the extreme edge it becomes the same as today - 5.

What it *might* break is anything that assumes the suffix is always 5 characters, but those things are arguably already broken, they just don't know it.